File: python/practicas/ejercicio_003.py
print("Este programa suma dos secuencias binarias que le pases acontinuacion")

seq1 = input("Ingresa la primera sequencia binaria: ")
seq2 = input("Ingresa la segunda secuencia binaria: ")

rdo_sum = 0

sequences = [seq1, seq2]

for seq in sequences:
    for i, bit in enumerate(reversed(seq)):
        if bit == "1":
            value_posicional = 2**i
            rdo_sum += value_posicional

print(f"El resultado de la suma de ambas secuencias binarias es: {rdo_sum}")

File: java/practicas/src/Ejercicio_003.java
import java.util.Scanner;

class Ejercicio_003 {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        String seq1, seq2;
        int rdo_sum = 0;

        System.out.println("Este programa suma dos secuencias binarias que pases acontinuacion");
        System.out.println("Ingrese la primera secuencia binaria: ");
        seq1 = input.nextLine();
        System.out.println("Ingrese la segunda secuencia binaria: ");
        seq2 = input.nextLine();

        String[] sequences = {seq1, seq2};

        for (String seq: sequences) {
            String reversed = new StringBuilder(seq).reverse().toString();
            for (int i=0; i < reversed.length(); i++) {
                if (reversed.charAt(i) == '1') {
                    rdo_sum += (int)Math.pow(2, i);
                }
            }
        }
        System.out.println("El resultado de la suma de ambas secuencias binarias es: "+rdo_sum);
    }
}